custom_components/rpi_gpio/cover.py

```diff
@@ -51,87 +59,116 @@
+        breaks_in_ha_version="2023.1.0",
+        is_fixable=False,
+        severity=IssueSeverity.WARNING,
+        translation_key="deprecated_yaml",
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up rpi_gpio cover."""
+    rpi_gpio: RpiGPIO = hass.data[DOMAIN][CONF_GPIO]
+    await hass.async_add_executor_job(rpi_gpio.setup_port, entry)
+
+    async_add_entities([RPiGPIOCover(hass, entry, rpi_gpio)], True)
 
 
 class RPiGPIOCover(CoverEntity):
     """Representation of a Raspberry GPIO cover."""
 
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
     def __init__(
-        self,
-        name,
-        relay_pin,
-        state_pin,
-        state_pull_mode,
-        relay_time,
-        invert_state,
-        invert_relay,
-        unique_id,
-    ):
-        """Initialize the cover."""
-        self._attr_name = name
-        self._attr_unique_id = unique_id
-        self._state = False
-        self._relay_pin = relay_pin
-        self._state_pin = state_pin
-        self._state_pull_mode = state_pull_mode
-        self._relay_time = relay_time
-        self._invert_state = invert_state
-        self._invert_relay = invert_relay
-        setup_output(self._relay_pin)
-        setup_input(self._state_pin, self._state_pull_mode)
-        write_output(self._relay_pin, 0 if self._invert_relay else 1)
-
-    def update(self):
-        """Update the state of the cover."""
-        self._state = read_input(self._state_pin)
+        self, hass: HomeAssistant, entry: ConfigEntry, rpi_gpio: RpiGPIO
+    ) -> None:
+        """Initialize the RPi GPIO entity."""
+        self.hass = hass
+        self.entry = entry
+        self.rpi_gpio = rpi_gpio
+        self.relay_pin: str = entry.data[CONF_RELAY_PIN]
+        self.state_pin: str = entry.data[CONF_STATE_PIN]
+        self._attr_unique_id = f"{self.relay_pin}-{self.state_pin}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._attr_unique_id)},
+            name=f"{entry.data[CONF_NAME]} (GPIOs {self._attr_unique_id})",
+            manufacturer="Raspberry Pi",
+        )
+
+    @property
+    def invert_relay(self) -> bool:
+        """Return if relay state should be inverted."""
+        return self.entry.options.get(CONF_INVERT_RELAY, DEFAULT_INVERT_RELAY)
 
     @property
-    def is_closed(self):
-        """Return true if cover is closed."""
-        return self._state != self._invert_state
+    def invert_state(self) -> bool:
+        """Return if state port should be inverted."""
+        return self.entry.options.get(CONF_INVERT_STATE, DEFAULT_INVERT_STATE)
 
-    def _trigger(self):
+    @property
+    def relay_time(self) -> float:
+        """Return the relay turn on duration."""
+        return self.entry.options.get(CONF_RELAY_TIME, DEFAULT_RELAY_TIME)
+
+    async def async_update(self) -> None:
```

Comment:
I think we need to call this first time as a last step in `__init__`

Reply:
This is already called when setting it to True
https://github.com/thecode/ha-rpi_gpio/blob/0d51d93541e1e7bb2fb21fba5beefa3f95199511/custom_components/rpi_gpio/cover.py#L89